### PR TITLE
Change auth id to mandatory

### PR DESCRIPTION
**Pull request by danield137**
trying to reduce some of the confusion working with app key and secret by change the kwarg to mandatory (no default value)

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/request.py [hunk 1]
```diff
@@ -96,13 +96,13 @@ def with_aad_user_password_authentication(cls, connection_string, user_id, passw
...
+    def with_aad_application_key_authentication(cls, connection_string, aad_app_id, app_key, authority_id):
         """Creates a KustoConnection string builder that will authenticate with AAD application and
         password.
         :param str connection_string: Kusto connection string should by of the format: https://<clusterName>.kusto.windows.net
         :param str aad_app_id: AAD application ID.
         :param str app_key: Corresponding password of the AAD application.
-        :param str authority_id: optional param. defaults to "common"
+        :param str authority_id: Authority id (aka Tenant id) must be provided 
         """
         _assert_value_is_valid(aad_app_id)
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/request.py]: Do we want to add:
_assert_value_is_valid(authority_id)
?

File: azure-kusto-data/tests/test_kusto_connection_string_builder.py [hunk 2]
```diff
@@ -31,25 +31,41 @@ def test_aad_app(self):
         uuid = str(uuid4())
         key = "key of application"
```
Comment (toshetah) [inline azure-kusto-data/tests/test_kusto_connection_string_builder.py]: Maybe we want to a test that expects an error when kcsb is initialized without an authority id.

File: azure-kusto-data/azure/kusto/data/request.py [hunk 3]
```diff
@@ -96,13 +96,13 @@ def with_aad_user_password_authentication(cls, connection_string, user_id, passw
         return kcsb
 
     @classmethod
-    def with_aad_application_key_authentication(cls, connection_string, aad_app_id, app_key, authority_id="common"):
+    def with_aad_application_key_authentication(cls, connection_string, aad_app_id, app_key, authority_id):
         """Creates a KustoConnection string builder that will authenticate with AAD application and
         password.
```
Comment (vladikbr) [inline azure-kusto-data/azure/kusto/data/request.py]: Please replace "password" with "key" here.

File: azure-kusto-data/tests/sample.py [hunk 4]
```diff
@@ -14,18 +14,22 @@
 client_id = "<insert here your AAD application id>"
 client_secret = "<insert here your AAD application key>"
 
-kcsb = KustoConnectionStringBuilder.with_aad_application_key_authentication(cluster, client_id, client_secret)
-# NOTE: it is possible to pass in an authority_id to all auth methods as well (defaults to common):
 # read more at https://docs.microsoft.com/en-us/onedrive/find-your-office-365-tenant-id
-# kcsb = KustoConnectionStringBuilder.with_aad_application_key_authentication(cluster, client_id, client_secret, authority_id)
+authority_id = "<insert here your tenet id>"
```
Comment (toshetah) [inline azure-kusto-data/tests/sample.py]: Typo

File: azure-kusto-data/tests/test_kusto_connection_string_builder.py [hunk 5]
```diff
@@ -31,25 +31,47 @@ def test_aad_app(self):
...
+            ),
+            KustoConnectionStringBuilder.with_aad_application_key_authentication(
+                "localhost", uuid, key, "microsoft.com"
             ),
-            KustoConnectionStringBuilder.with_aad_application_key_authentication("localhost", uuid, key),
         ]
 
+        try:
+            KustoConnectionStringBuilder.with_aad_application_key_authentication("localhost", uuid, key, None)
+        except Exception as e:
```
Comment (toshetah) [inline azure-kusto-data/tests/test_kusto_connection_string_builder.py]: except ValueError as e:

**Pull request merged**